### DISCUSSION TITLE
Observed `Object Not Found Matching Id` error in Sentry

### DISCUSSION THREAD

**Discussion by Lohins** [cefsharp/CefSharp#3633] [General]
### Title
Observed the following errors in our Sentry logs.

```
UnhandledRejection
Non-Error promise rejection captured with value: Object Not Found Matching Id:2
```

### Description
First of all, we are not using CefSharp directly.
But we observed the following errors in our Sentry logs.

```
UnhandledRejection
Non-Error promise rejection captured with value: Object Not Found Matching Id:2
```

And looks like the message comes from `TryCallMethodResult` exception, [reference](https://github.com/cefsharp/CefSharp/blob/445fb9ab2d178b55e3ca6561722d151ce3d9337e/CefSharp/Internals/JavascriptObjectRepository.cs#L293)

Our logs show thoses errors was coming from Microsoft company, so it is highly likely that Microsoft did some SafeLink Scanning on emails(we attached our websites in emails).

### What is the expectation? What do you see instead?
**Expectation**: no seeing errors out of our application, for instance from CefSharp.
**Actual**: seeing errors from CefSharp `TryCallMethodResult`

### Additional information
Our website has cross-domain redirects, will this invalidate the JavascriptObject repository, then cause that error?

.

**Comment by amaitland** [comment]
> First of all, we are not using CefSharp directly.
But we observed the following errors in our Sentry logs.

Please contact the developer and have them complete the bug report temple available at https://github.com/cefsharp/CefSharp/blob/master/.github/ISSUE_TEMPLATE/bug_report.md#bug-report

**Reply by Alvaro-c** [reply]
Hi [amaitland](https://github.com/amaitland) and thank you for your help, 

I tried to fill the Bug report, but I am missing almost all the information required in the template. 

Long story short: 

1. Sentry (https://sentry.io/) is reporting the error described above
2. It seems that it occurs when outlook check links in emails
3. The error message match with a CefSharp error from [this file ](https://github.com/cefsharp/CefSharp/blob/445fb9ab2d178b55e3ca6561722d151ce3d9337e/CefSharp/Internals/JavascriptObjectRepository.cs#L293) mentioned above
4. Sentry users do not have any info to fill the bug report

I opened this discussion on Sentry's side: https://github.com/getsentry/sentry/discussions/36121

Is there any other way to report the issue, apart of filling the template? 
I would prefer to avoid reporting it not following the procedure you specify

Thanks a lot !

**Reply by amaitland** [reply]
The sentry devs will need to fill out the bug report. 

For all I know this is a bug in sentry. 

Without an example that clearly reproduces the problem there's nothing I can do.

**Reply by Alvaro-c** [reply]
Thank you very much for your answer, I will link your comment in the conversation I have opened in Sentry's side.

**Reply by lobsterkatie** [reply]
Hi, there - Sentry dev here. I similarly don't have enough information to fill in most of the template. 

@Alvaro-c - What information were you missing when you tried to file the bug? Perhaps I can help fill in those blanks?

@amaitland - Similar question. Bottom line, we're an error-reporting service, so if cefsharp is throwing an error, we'll pick it up. AFAICT, the error has been adequately described above, so what is it you're hoping to know from our end? Happy to help here, but I'm not yet understanding how we're anything but the messenger in this equation. Perhaps this would be helpful background? https://github.com/getsentry/sentry-javascript/issues/3440

**Reply by amaitland** [reply]
>  Similar question. Bottom line, we're an error-reporting service,

@lobsterkatie Thanks for the clarification, it wasn't clear to be that sentry was an error reporting service. I've never heard of the product.

> Happy to help here, but I'm not yet understanding how we're anything but the messenger in this equation

Thanks, that's clearer now. I'm not familiar with your product so I'm not sure what information you can provide.

From a `CefSharp` point of view commit https://github.com/cefsharp/CefSharp/commit/2ead42338062ec60c9bfc8564b88fb4307dc7bc3 will add some additional error information to the error message that should help anyone using `CefSharp` directly better identify the method that that was causing a problem. Past that I'm not really sure what else I can add here, as yet I've not received enough information to consider this a bug, it's just as likely user error.

**Comment by Alvaro-c** [comment]
Hello all, 

Thank you [amaitland](https://github.com/amaitland) and [lobsterkatie](https://github.com/lobsterkatie) for your help in this one. 

As quick summary, it seems that we all three are in a position with little to no power here. 

From our side, I am just receiving the report of an `Exception` that does not occur on my side
From `Sentry` perspective they are raising an error that they receive, raised by Outlook´s crawlers
From `CefSharp` they are managing a possible `Exception`  that can happen when using their code. 

It is Outlook team who is using `CefSharp` directly, therefore it seems that only them could provide information to fill a bug report, if the issue is not in their end.

I will try to find the way to report this issue to them. 

Again, thank you so much for your help.

**Reply by lobsterkatie** [reply]
> It is Outlook team who is using `CefSharp` directly, therefore it seems that only them could provide information to fill a bug report, if the issue is not in their end.
> 
> I will try to find the way to report this issue to them.

Great, thanks, @Alvaro-c. Please let us know how you make out!

**Reply by edgariscoding** [reply]
Any update on this? We've seen random spikes, up to 3,400 events in the span of 15 minutes for this error. I guess we'll be ignoring the issue but it would have been nice for Outlook to fix the issue themselves.

**Reply by Alvaro-c** [reply]
Unfortunately, no updates on my side. I tried to contact Outlook team back in the day, but if I remember correctly, I only found usual general support channels and I couldn't reach any useful response.

**Reply by edgariscoding** [reply]
Ah thank you anyway! 

I also just found out that these CefSharp errors we're receiving were actually caused by AOL Desktop Gold rather than Outlook, so it looks like there's plenty of other applications causing this exact issue. I would have imagined that CefSharp should be the one to help them address this issue or fix it on their end since it seems unreasonable to expect end users to to contact all the software companies that use this package.
